portalnet/src/socket.rs

```diff
@@ -1,10 +1,20 @@
-use std::net::{SocketAddr, UdpSocket};
+use core::time;
+use std::{
+    net::{IpAddr, SocketAddr, TcpStream, UdpSocket},
+    thread,
+};
 use tracing::{debug, info, warn};
 
 // This stun server is part of the testnet infrastructure.
 // If you are unable to connect, please create an issue.
 const STUN_SERVER: &str = "159.223.0.83:3478";
 
+/// The duration in seconds of an external port mapping by UPnP.
+const UPNP_MAPPING_DURATION: u32 = 3600;
```

Comment:
Why are we mapping UPnP for only 60 minutes? I think we want to use an infinite value here, which in this case should be 0.

Reply:
I am slightly in the camp of having a limit unless we're absolutely sure that these will expire after a period of disuse.  We don't want to be opening ports up on people's home networks indefinitely.  Seems worth the effort to implement periodic renewal of the mapping to make sure we aren't being bad citizens on user machines.

Reply:
> I am slightly in the camp of having a limit unless we're absolutely sure that these will expire after a period of disuse. We don't want to be opening ports up on people's home networks indefinitely. Seems worth the effort to implement periodic renewal of the mapping to make sure we aren't being bad citizens on user machines.

Agree.  The current design will initially map the port for 60 minutes, and then periodically (every 30 minutes) renew the port for another 60 minutes.  That said if Trin is down for no matter what reasons (crash or gracefully closed), the port reserved by Trin will be automatically released in < 60 minutes.